Vendor/SocketIO/SocketIO.m

```diff
@@ -163,7 +163,8 @@ - (void) connectToHost:(NSString *)host
         }
         else {
             // connection failed
-            [self connection:_handshake didFailWithError:nil];
```

Comment:
@ashar02 While I totally understand that compile warnings bothers you, I am up to fix our own code compile warnings and for third party libraries either update the library or leave things as they are. I don't think is good for us to modify code vendors inside our own repo, perhaps we need to link with those vendors in some other way.

I remember one incident about OpenSSL that was caused by removing compiler warnings and it lead to a several security hole, I am sure this will not be the case but I think it probes a point. Here is a short blog post in the matter of the OpenSSL issue: https://www.schneier.com/blog/archives/2008/05/random_number_b.html